ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/TestIdsGenerator.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.apitesting
+
+import scala.util.Random
+
+/**
+  * Provides a mechanism to randomize various identifiers used in testing per run of a process.
+  *
+  * It enables Ledger API Test Tool to be ran against a persistent Ledger API endpoint repeatedly by avoiding
+  * duplication of identifiers.
+  */
+trait TestIdsGenerator {
+  self: MultiLedgerFixture =>
+
+  lazy val runSuffix = Random.alphanumeric.take(10).mkString
+  lazy val runCommandSuffix = if (config.uniqueCommandIdentifiers) "-" + runSuffix else ""
+  // Using runCommandSuffix, since there does not seem to be a need to treat workflow ids differently than command ids.
+  // Yet lets wrap those identifiers into an appropriate function so we could do it in the future.
+  lazy val runWorkflowSuffix = runCommandSuffix
+  lazy val runPartySuffix = if (config.uniquePartyIdentifiers) "-" + runSuffix else ""
+  def partyNameUnifier(partyText: String) = partyText + runPartySuffix
```

Comment:
It's not clear what this function "unifies". Maybe drop the `Unifier` suffix or rename it to `testPartyName`, etc...?

Reply:
Yes, I was also not happy with the name. `testPartyName` sounds so much better!